graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/relay/PaginationHelper.java

```diff
@@ -154,10 +164,27 @@ private static <T> List<T> collectItems(Iterator<T> it, Arguments arguments, Cur
             if (arguments.before != null && cursorSupport.getCursor(value).equals(arguments.before)) {
                 break;
             }
+
+            //Adding a limit of 1000 to avoid OOM
+            if(count.intValue() == 500) {
+                logger.warn("The current paginated query is returning more than 500 items. This may cause a memory leak.");
+            } else if(count.intValue() == nodeLimit && logger.isWarnEnabled()) {
+                logger.warn("The current paginated query is returning more than {} items. Stopping the query.", nodeLimit, new DataFetchingException("The current paginated query is returning more than " + nodeLimit + " items. Stopping the query here."));
```

Comment:
May be the Exception would be better to be print in dedub level ?
like suggesting to activate debug log level to have more informations, and in debug level log the stack.
Just scared that in case a customer have this limit reach for any reasons it's gonna log a lot